modules/db/src/main/scala/gem/dao/ObservationDao.scala

```diff
@@ -21,6 +21,7 @@ object ObservationDao {
                     ${o.id.index},
                     ${o.title},
                     ${o.instrument})
+       ON CONFLICT DO NOTHING
```

Comment:
I think I would just do the insert and catch the duplicate key exception. Something like

```scala
import doobie.contrib.postgresql.syntax._

doInsert(...).onUniqueViolation(0.point[ConnectionIO]) // ConnectionIO[Int]
```